File: Programmers/Lv.1/신고 결과 받기/92334.js
function solution(id_list, report, k) {
    const answer = new Array(id_list.length).fill(0);
    
    const reportMap = new Map();
    const countMap = new Map();
    id_list.forEach(id => {
        reportMap.set(id, []);
        countMap.set(id, 0);
    });
    
    const reportSet = new Set(report);
    reportSet.forEach(rs => {
        const r = rs.split(' ');
        reportMap.get(r[0]).push(r[1]);
        countMap.set(r[1], countMap.get(r[1]) + 1);
    });
    
    id_list.forEach((v, i) => {
       countMap.forEach((value, key) => {
           if (value >= k && reportMap.get(v).includes(key)) {
               answer[i]++;
           }
       });
    });
    
    return answer;
}

File: Programmers/Lv.1/신고 결과 받기/92334.java
import java.util.*;

class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        int[] answer = new int[id_list.length];
        
        Map<String, List<String>> reportMap = new HashMap<>();
        Map<String, Integer> countMap = new HashMap<>();
        for (String id : id_list) {
            List<String> list = new ArrayList<>();
            reportMap.put(id, list);
            countMap.put(id, 0);
        }
        
        Set<String> reportSet = new HashSet<>(Arrays.asList(report));
        Iterator<String> iter = reportSet.iterator();
        while (iter.hasNext()) {
            String[] next = iter.next().split(" ");
            reportMap.get(next[0]).add(next[1]);
            countMap.put(next[1], countMap.get(next[1]) + 1);
        }
        
        for (int i = 0; i < id_list.length; i++) {
            for (String key : countMap.keySet()) {
                if (countMap.get(key) >= k && reportMap.get(id_list[i]).contains(key)) {
                    answer[i]++;
                }
            }
        }
        
        return answer;
    }
}
